### PR TITLE
Chapter 6 Modelling User

### PR DESCRIPTION
**Pull request by daniadore**
Ready

### CODE REVIEW

**Comment by daniadore** [comment]
Ready

File: test/fixtures/users.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (ngocvt-0484) [inline test/fixtures/users.yml:3]: don't need do test
please delete it

File: test/models/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (ngocvt-0484) [inline test/models/user_test.rb:6]: don't need do test
please delete it

**Comment by daniadore** [comment]
Ready

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save {self.email = email.downcase}
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```
Comment (anlvq-1058) [inline app/models/user.rb:6]:
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i

  before_save {self.email = email.downcase}

  validates :name, presence: true
  validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```

File: db/migrate/20210825063235_dropusers.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Dropusers < ActiveRecord::Migration[6.1]
+  def change
+    drop_table :users
```
Comment (anlvq-1058) [inline db/migrate/20210825063235_dropusers.rb:3]: why you need to drop the table users when you even create it?

**Review by anlvq-1058 — COMMENTED** [review]
cmt

File: db/migrate/20210825063235_dropusers.rb (hunk 4 above)
Comment (daniadore) [reply db/migrate/20210825063235_dropusers.rb:3]: i also don't understand.... so what should i do ? :D

**Comment by daniadore** [comment]
Ready

File: db/migrate/20210825063235_dropusers.rb (hunk 4 above)
Comment (anlvq-1058) [reply db/migrate/20210825063235_dropusers.rb:3]: You should delete this file, it make you code confuse

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by daniadore** [comment]
Ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**